server/routes/choropleth.py

```diff
@@ -35,85 +35,109 @@
 
 @bp.route('/download')
 def download():
-    """Returns data for all countries for a certain statistical variable.
+    """Returns data for geographic subregions for a certain statistical 
+            variable.
 
     API Params:
-        statVar -> The statistical variable to download.
-        perCapita -> Whether to return the per-capita value.
-        place -> The currently viewed geography to render.
+        statVar -> The statistical variable to download, as a string.
+        perCapita -> Whether to return the per-capita value, as a string of a 
+                boolean.
+        geoId -> The currently viewed geography to render.
+        level -> The subgeographic level to pull and display information for.
+        mdom -> The measurement denominator to use if perCapita is true. 
+                Defaults to Count_Person.
+
 
     API Returns:
-        df -> dictionary of statistical variable values by country.
-            Additionally includes some plotting information.
+        df -> Returns a dictionary with two entries
+            geoJson -> geoJson format that includes statistical variables info,
+                    geoId, and name for all subregions.
+            _PLOTTING_INFO -> Includes domain range for use in D3 and the
+                    color palette to render with.
     """
     # Get required request parameters.
-    stat_var_requested = request.args.get("statVar")
-    if stat_var_requested is None:
+    requested_stat_var = request.args.get("statVar")
+    if requested_stat_var is None:
         return jsonify({"error": "Must provide a 'statVar' field!"}, 400)
-    place_requested = request.args.get("place")
-    if place_requested is None:
-        return jsonify({"error": "Must provide a 'place' field!"}, 400)
+    requested_geoId = request.args.get("geoId")
+    if requested_geoId is None:
+        return jsonify({"error": "Must provide a 'geoId' field!"}, 400)
 
     # Get optional fields.
-    per_capita = request.args.get("perCapita", default=False) == "true"
-
-    # Get the type of the requested place. Accept first valid type.
-    geolevel_below = dc.get_property_values([place_requested],
-                                            "typeOf")[place_requested]
-    for level in geolevel_below:
-        if level in LEVEL_MAP:
-            geolevel_below = level
+    per_capita = request.args.get("perCapita", default=False) in (
+        [True, "true", "True", 1, "t"])
```

Comment:
```suggestion
    per_capita = request.args.get("perCapita", default=False).lower() in (
        ["true", "1", "t"])
```

I am not sure if this is correct.  I do not know the type that request.args.get returns.  I cannot find documentation.  My suggestion assumes only a string is returned.

Reply:
Changed.